packages/core/components/EditMetadata/NewAnnotationPathway.tsx

```diff
@@ -18,21 +31,87 @@ enum EditStep {
+    const { onDismiss, hasUnsavedChanges } = props;
+
     const [step, setStep] = React.useState<EditStep>(EditStep.CREATE_FIELD);
-    const [newValues, setNewValues] = React.useState<string | undefined>();
+    const [newValues, setNewValues] = React.useState<AnnotationValue | undefined>();
     const [newFieldName, setNewFieldName] = React.useState<string>("");
+    const [newFieldDescription, setNewFieldDescription] = React.useState<string>("");
     const [newFieldDataType, setNewFieldDataType] = React.useState<AnnotationType | undefined>();
     const [newDropdownOption, setNewDropdownOption] = React.useState<string>("");
     const [dropdownOptions, setDropdownOptions] = React.useState<IComboBoxOption[]>([]);
+    const [submissionStatus, setSubmissionStatus] = React.useState<AnnotationStatus | undefined>();
+
+    const statuses = useSelector(interaction.selectors.getProcessStatuses);
+    const annotationCreationStatus = React.useMemo(
+        () => statuses.find((status) => status.processId === newFieldName),
+        [newFieldName, statuses]
+    );
+    // Check for updates to the annotation submission status
+    React.useEffect(() => {
+        const checkForStatusUpdates = async () => {
+            const currentStatus = annotationCreationStatus?.data?.status;
+            switch (currentStatus) {
+                case ProcessStatus.ERROR:
+                case ProcessStatus.STARTED:
+                case ProcessStatus.PROGRESS:
+                    setSubmissionStatus({
+                        status: currentStatus,
+                        message: annotationCreationStatus?.data?.msg,
+                    });
+                    return;
+                case ProcessStatus.SUCCEEDED:
+                    if (newFieldName && newValues) {
+                        try {
+                            dispatch(
+                                interaction.actions.editFiles({ [newFieldName]: [newValues] })
+                            );
+                        } catch (e) {
+                            setSubmissionStatus({
+                                status: ProcessStatus.ERROR,
+                                message: `Failed to create annotation: ${e}`,
+                            });
+                        } finally {
+                            hasUnsavedChanges(false);
+                            onDismiss();
+                        }
+                    }
+                default:
+                    return;
+            }
+        };
+        checkForStatusUpdates();
+    }, [annotationCreationStatus, dispatch, hasUnsavedChanges, newFieldName, newValues, onDismiss]);
+
+    const onChangeAlphanumericField = (
+        e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+        newValue: string | undefined
+    ) => {
+        const regex = /^[\w\-\s]+$/g;
+        // Restricts character entry to alphanumeric
+        if (newValue && !regex.test(newValue)) {
```

Comment:
aww, no injection attacks?  But what if scientists want to mark an image ";Drop tables;":True?